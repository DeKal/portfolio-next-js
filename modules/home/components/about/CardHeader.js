import * as React from 'react'
import styled from 'styled-components/macro'
import { Translate, withLocalize } from 'react-localize-redux'

const CardHeader = () => (
  <Header>
    <MainText>
      <Translate id="About me" />
    </MainText>
    <SubText>
      <Translate id="About Me: Subtext" />
    </SubText>
  </Header>
)

export default withLocalize(CardHeader)

const Header = styled.header`
  padding: 20px;
  background-color: ${props => props.theme.colors.colorAccent};
`

const MainText = styled.h2`
  color: ${props => props.theme.colors.colorLight};
`

const SubText = styled.h4`
  font-style: italic;
`
