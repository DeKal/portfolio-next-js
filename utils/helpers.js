import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
