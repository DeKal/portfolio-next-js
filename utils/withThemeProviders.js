import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
