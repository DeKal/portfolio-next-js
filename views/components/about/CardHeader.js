import * as React from 'react'
import styled from 'styled-components/macro'

const CardHeader = () => (
  <Header>
    <MainText>About me</MainText>
    <SubText>Well skip it if you&apos;re bored!!</SubText>
  </Header>
)

export default CardHeader

const Header = styled.header`
  padding: 20px;
  background-color: ${props => props.theme.colors.colorAccent};
`

const MainText = styled.h2`
  color: ${props => props.theme.colors.colorLight};
`

const SubText = styled.h4`
  /* no style */
`
