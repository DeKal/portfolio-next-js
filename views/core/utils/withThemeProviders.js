import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
