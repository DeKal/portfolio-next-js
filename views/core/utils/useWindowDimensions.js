import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
