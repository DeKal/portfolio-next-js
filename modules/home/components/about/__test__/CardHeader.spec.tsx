import * as React from 'react'
import { mountWithTheme } from '~/modules/core/utils/withThemeProviders'
import CardHeader from '~/modules/home/components/about/CardHeader'

describe('Card Header', () => {
  it('should render without throwing an error', function() {
    const wrap = mountWithTheme(<CardHeader />)
    expect(wrap).toMatchSnapshot()
  })
})
