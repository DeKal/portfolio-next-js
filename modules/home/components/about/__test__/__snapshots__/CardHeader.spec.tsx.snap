// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Header should render without throwing an error 1`] = `
<LocalizedComponent>
  <CardHeader
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    languages={Array []}
    renderToStaticMarkup={false}
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <CardHeader__Header>
      <header
        className="CardHeader__Header-r5umbs-0 hfwWjn"
      >
        <CardHeader__MainText>
          <h2
            className="CardHeader__MainText-r5umbs-1 EBlPf"
          >
            <Translate
              id="About me"
            >
              <WrappedTranslate
                context={
                  Object {
                    "activeLanguage": undefined,
                    "addTranslation": [Function],
                    "addTranslationForLanguage": [Function],
                    "defaultLanguage": undefined,
                    "ignoreTranslateChildren": false,
                    "initialize": [Function],
                    "languages": Array [],
                    "renderToStaticMarkup": false,
                    "setActiveLanguage": [Function],
                    "translate": [Function],
                  }
                }
                id="About me"
              >
                Missing translationId: About me for language: \${ languageCode }
              </WrappedTranslate>
            </Translate>
          </h2>
        </CardHeader__MainText>
        <CardHeader__SubText>
          <h4
            className="CardHeader__SubText-r5umbs-2 kzjOLC"
          >
            <Translate
              id="About Me: Subtext"
            >
              <WrappedTranslate
                context={
                  Object {
                    "activeLanguage": undefined,
                    "addTranslation": [Function],
                    "addTranslationForLanguage": [Function],
                    "defaultLanguage": undefined,
                    "ignoreTranslateChildren": false,
                    "initialize": [Function],
                    "languages": Array [],
                    "renderToStaticMarkup": false,
                    "setActiveLanguage": [Function],
                    "translate": [Function],
                  }
                }
                id="About Me: Subtext"
              >
                Missing translationId: About Me: Subtext for language: \${ languageCode }
              </WrappedTranslate>
            </Translate>
          </h4>
        </CardHeader__SubText>
      </header>
    </CardHeader__Header>
  </CardHeader>
</LocalizedComponent>
`;
